ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/PersistDataStep.java

```diff
@@ -85,6 +72,19 @@ public List<Task<SnapDataRequest>> persist(final List<Task<SnapDataRequest>> tas
               downloadState.getMetricsManager().notifyNodesGenerated(persistedNodes);
             }
           }
+
+          // enqueue child requests
+          final Stream<SnapDataRequest> childRequests =
```

Comment:
we need a specific behavior for the heal

Reply:
can you expand on this? Are you saying that moving the children enqueuing after persisting the nodes somehow invalidate the children of the task?